### DISCUSSION TITLE
Move __layouts to their own folder like params

### DISCUSSION THREAD

**Discussion by homerjam** [sveltejs/kit#4583] [Ideas]
With the named layout syntax available how about moving the layouts to their own folder and having a reserved layout named 'default'.

Secondary request - find a nicer way to run arbitrary code on every page load for things like auth and setting up a urql that doesn't involve repetition/making a mess in layouts.

**Comment by stephane-vanraes** [comment]
This would introduce unnecessary _noise_ in filenames, if you do that you would get a folder structure like:

```
/products
  index@product.svelte
  [id]@product.svelte
  search@product.svelte
  request@product.svelte
```

instead of the easier to process
```
/products
  __layout.svelte
  index.svelte
  [id].svelte
  search.svelte
  request.svelte
```

and if you get nested layouts this would become even more unwieldly.

**Comment by homerjam** [comment]
Good point. Is it possible to do this?

```
/products@product
  index.svelte
  [id].svelte
  search.svelte
  request.svelte
```

**Reply by stephane-vanraes** [reply]
Not currently no.
Don't think there any plans to add it either

**Comment by manvankaliber** [comment]
They added named layouts recently. I also suggested it in https://github.com/sveltejs/kit/discussions/5037
